src/libcore/ptr/mod.rs

```diff
@@ -248,14 +246,17 @@ pub(crate) struct FatPtr<T> {
 ///
 /// // create a slice pointer when starting out with a pointer to the first element
 /// let x = [5, 6, 7];
-/// let ptr = x.as_ptr();
-/// let slice = ptr::slice_from_raw_parts(ptr, 3);
+/// let raw_pointer = x.as_ptr();
+/// let slice = ptr::slice_from_raw_parts(raw_pointer, 3);
 /// assert_eq!(unsafe { &*slice }[2], 7);
 /// ```
 #[inline]
 #[stable(feature = "slice_from_raw_parts", since = "1.42.0")]
 #[rustc_const_unstable(feature = "const_slice_from_raw_parts", issue = "67456")]
 pub const fn slice_from_raw_parts<T>(data: *const T, len: usize) -> *const [T] {
+    // SAFETY: Creating the Repr is safe, but using it is unsafe, hence the
+    // `unsafe` block. Using the return value can lead to undefined behavior
+    // as explained in the documentation for [`from_raw_parts`]
```

Comment:
The unsafe operation here is to get the `rust` value out of the union `Repr`. I think is safe because `*const [T]` has the same memory layout as `FatPtr`

Reply:
Fixed in ef32ab4